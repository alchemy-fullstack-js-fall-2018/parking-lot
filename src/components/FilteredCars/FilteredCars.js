import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import Cars from '../../components/Cars/Cars';

export default class FilteredCars extends PureComponent {
  static propTypes = {
    cars: PropTypes.array.isRequired,
    searchTerm: PropTypes.string.isRequired,
    updateSearchTerm: PropTypes.func.isRequired,
    summary: PropTypes.string
  };

  render() {
    const { cars, searchTerm, updateSearchTerm, summary } = this.props;
    return (
      <Fragment>
        <form>
          <label htmlFor="searchTerm">Search: </label>
          <input type="text" value={searchTerm} onChange={updateSearchTerm} />
        </form>
        <Cars cars={cars} summary={summary}/>
      </Fragment>
    );
  }
}
