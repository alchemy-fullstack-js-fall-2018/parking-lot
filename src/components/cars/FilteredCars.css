.header {
  text-align: center;
}

.form {
  display: block;
  text-align: center;
}