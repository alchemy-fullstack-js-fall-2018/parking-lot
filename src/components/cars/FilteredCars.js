import React, { Fragment } from 'react';
import styles from './FilteredCars.css';
import Cars from '../../components/cars/Cars';

export default function FilteredCars({ cars, searchTerm, updateSearchTerm }) {
  return (
    <Fragment>
      <h3 className={styles.header}>Search Cars</h3>
      <div>
        <form className={styles.form}>
          <label htmlFor="searchTerm">Search: </label>
          <input type="text" value={searchTerm} onChange={updateSearchTerm} />
        </form>
      </div>
      <Cars cars={cars} />
    </Fragment>
  );
}
