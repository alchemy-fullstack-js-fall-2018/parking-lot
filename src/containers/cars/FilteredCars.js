import { connect } from 'react-redux';
import FilteredCars from '../../components/cars/FilteredCars';
import { updateCarSearchTerm } from '../../actions/cars';
import { getFilteredCars, getCarsSearchTerm } from '../../selectors/cars';


const mapStateToProps = state => ({
  cars: getFilteredCars(state),
  searchTerm: getCarsSearchTerm(state)
});

const mapDispatchToProps = dispatch => ({
  updateSearchTerm: ({ target }) => dispatch(updateCarSearchTerm(target.value))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FilteredCars);
