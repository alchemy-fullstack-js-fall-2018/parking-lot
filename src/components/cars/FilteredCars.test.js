import React from 'react';
import { shallow } from 'enzyme';
import FilteredCars from './FilteredCars';

describe('Filtered Cars', () => {
  let wrapper;
  beforeEach(() => {
    const cars = [{
      make: 'a make',
      model: 'a model',
      plate: 'a plate'
    }];
    const searchTerm = 'a search term';
    const updateSearchTerm = jest.fn();
    wrapper = shallow(<FilteredCars cars={cars} searchTerm={searchTerm} updateSearchTerm={updateSearchTerm}/>);
  });
  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
