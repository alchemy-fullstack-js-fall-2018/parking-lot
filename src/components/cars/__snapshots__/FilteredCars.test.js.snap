// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filtered Cars renders 1`] = `
<Fragment>
  <h3
    className="header"
  >
    Search Cars
  </h3>
  <div>
    <form
      className="form"
    >
      <label
        htmlFor="searchTerm"
      >
        Search: 
      </label>
      <input
        onChange={[MockFunction]}
        type="text"
        value="a search term"
      />
    </form>
  </div>
  <Cars
    cars={
      Array [
        Object {
          "make": "a make",
          "model": "a model",
          "plate": "a plate",
        },
      ]
    }
  />
</Fragment>
`;
